src/directives/when.ts

```diff
@@ -12,91 +12,63 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import {directive, Directive, NodePart, reparentNodes} from '../lit-html.js';
+import {directive, Directive, NodePart} from '../lit-html.js';
 
-interface PartCache {
-  truePart: NodePart;
-  falsePart: NodePart;
-  prevCondition?: boolean;
-  cacheContainer: DocumentFragment;
-}
-
-const partCaches = new WeakMap<NodePart, PartCache>();
+export type WhenValue = () => any;
+export type CaseMap = {
+  [key: string]: WhenValue
+};
 
 /**
- * Efficiently switches between two templates based on the given condition. The
- * rendered content is cached, and re-used when switching conditions. Templates
- * are evaluated lazily, so the passed values must be functions.
- *
- * While this directive can render any regular part, it makes the most sense
- * when used with TemplateResult since most other values are dirty checked
- * already.
+ * Directive for handling conditional logic inside templates. One or
+ * two function parameters are treated as if or if/else statements. The
+ * first parameter is checked against truthiness and the associated value
+ * is rendered.
  *
- * Example:
+ * An object is treated as a case switch, where the condition is used as
+ * key to render the value of the matched case on the object. If no condition
+ * matches the default case is rendered if present. Keys can be strings and
+ * symbols.
  *
- * ```
- * let checked = false;
+ * Templates are re-instantiated each re-render. For caching nodes between
```

Comment:
This is not exactly correct, templates are not always re-instantiated on re-render. This directive uses the regular lit-html template handling, which re-uses templates by default. Rendering the following two things is effectively the same, caching-wise:

```javascript
when(true, html`<div></div>`)
// Same as
html`<div></div>`
```

Oonly switching between templates can cause template re-instantiation, as old templates will not be cached. This means you only re-instantiate a template when you switch between templates in an A -> B -> A fashion.

Reply:
You're right, I will adjust the phrasing.